go/internal/feast/sqliteonlinestore.go

```diff
@@ -0,0 +1,159 @@
+	serialized_entities := make([]interface{}, len(entityKeys))
+	for i := 0; i < len(entityKeys); i++ {
+		serKey, err := serializeEntityKey(&entityKeys[i])
+		if err != nil {
+			return nil, err
+		}
+		// TODO: fix this, string conversion is not safe
+		entityNameToEntityIndex[string(*serKey)] = i
+		// for IN clause in read query
+		in_query[i] = "?"
+		serialized_entities[i] = *serKey
+	}
+	featureNamesToIdx := make(map[string]int)
+	for idx, name := range featureNames {
+		featureNamesToIdx[name] = idx
+	}
+
+	for idx := 0; idx < len(entityKeys); idx++ {
+		results[idx] = make([]FeatureData, featureCount)
+	}
+	for _, featureViewName := range featureViewNames {
+		query_string := fmt.Sprintf(`SELECT entity_key, feature_name, value, event_ts
+									FROM %s
+									WHERE entity_key IN (%s)
+									ORDER BY entity_key`, tableId(project, featureViewName), strings.Join(in_query, ","))
+		rows, err := s.db.Query(query_string, serialized_entities...)
+		if err != nil {
+			return nil, err
+		}
+		defer rows.Close()
+		for rows.Next() {
+			var entity_key string
+			var feature_name string
+			var valueString string
+			var event_ts time.Time
+			var value types.Value
+			err = rows.Scan(&entity_key, &feature_name, &valueString, &event_ts)
+			if err != nil {
+				log.Fatal(err)
+			}
+			if err := proto.Unmarshal([]byte(valueString), &value); err != nil {
+				return nil, errors.New("error converting parsed value to types.Value")
+			}
+			results[entityNameToEntityIndex[entity_key]][featureNamesToIdx[feature_name]] = FeatureData{reference: serving.FeatureReferenceV2{FeatureViewName: featureViewName, FeatureName: feature_name},
+				timestamp: *timestamppb.New(event_ts),
+				value:     types.Value{Val: value.Val},
+			}
+		}
+	}
+	return results, nil
+}
+
+func (s *SqliteOnlineStore) Update(ctx context.Context, config *RepoConfig, tables_to_delete []*FeatureView, tables_to_keep []*FeatureView) error {
+	_, err := s.getConnection()
+	if err != nil {
+		return err
+	}
+	project := config.Project
+	for _, table := range tables_to_keep {
+		s.db.Exec(
+			fmt.Sprintf("CREATE TABLE IF NOT EXISTS %s (entity_key BLOB, feature_name TEXT, value BLOB, event_ts timestamp, created_ts timestamp,  PRIMARY KEY(entity_key, feature_name))", tableId(project, table.base.name)))
+		s.db.Exec(
+			fmt.Sprintf("CREATE INDEX IF NOT EXISTS %s_ek ON %s (entity_key);", tableId(project, table.base.name), tableId(project, table.base.name)))
+	}
+	for _, table := range tables_to_delete {
+		s.db.Exec("DROP TABLE IF EXISTS %s", tableId(project, table.base.name))
+	}
+	return nil
+}
+
+// Gets a sqlite connection and sets it to the online store and also returns a pointer to the connection.
+func (s *SqliteOnlineStore) getConnection() (*sql.DB, error) {
+	if s.db == nil {
+		if s.path == "" {
+			return nil, errors.New("no database path available")
+		}
+		db, err := initializeConnection(s.path)
+		s.db = db
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
This should probably be guarded by a lock to prevent concurrent access creating multiple dangling connections

Reply:
added a mutex.